go/libkb/sig_chain.go

```diff
@@ -422,21 +442,20 @@ func (sc *SigChain) verifySubchain(ctx context.Context, kf KeyFamily, links []*C
 
 		if first {
 			if err = ckf.InsertEldestLink(tcl, un); err != nil {
-				return
+				return cached, cki, err
 			}
 			first = false
 		}
 
-		// Optimization: When several links in a row are signed by the same
-		// key, we only validate the signature of the last of that group.
-		// (Unless a link delegates new keys, in which case we always check.)
+		// Optimization: only check sigs on some links, like the final
+		// link, or those that delegate and revoke keys.
 		// Note that we do this *before* processing revocations in the key
 		// family. That's important because a chain link might revoke the same
 		// key that signed it.
 		isDelegating := (tcl.GetRole() != DLGNone)
 		isModifyingKeys := isDelegating || tcl.Type() == DelegationTypePGPUpdate
 		isFinalLink := (linkIndex == len(links)-1)
-		isLastLinkInSameKeyRun := (isFinalLink || newKID != links[linkIndex+1].GetKID())
+		hasRevocations := link.HasRevocations()
```

Comment:
NB: Removing the logic that checks at the end of a "run" of links all signed by the same key. It was impossible to determine without more information in the V2 outer links, and I don't think it really buys us much securitywise. I Think we can get away with only checking the last sig, honestly, so this seems fine.